### ISSUE TITLE
[v2 QUESTION]: Training fails using RDKit molecule featurizers 

### ISSUE THREAD

**Issue by tamarahalter** [chemprop/chemprop#1138]
Hey!

I want to use chemprop to train a model for predicting logD values. I have already successfully trained models with chemprop but have an issue with the newest version v.2.1 and the RDKit molecule featurizers. When I train a model without a featurizer everything works fine, my code also works with the MorganBinaryFeaturizer. However, when I use any of the RDKit featurizers I get val_loss=nan.0 during the training and the model then also only predicts NaNs. 

This is the code im currently using for the model training:
```
import pandas as pd
import numpy as np
from pathlib import Path
from lightning import pytorch as pl
from chemprop import data, featurizers, models, nn, utils

smiles_column = 'smiles'
target_column = ['logD']

df_input = pd.read_csv(data_dir / '1_all_logD.csv')

smis = df_input.loc[:, "smiles"].values
ys = df_input.loc[:, target_column].values

mols = [utils.make_mol(smi, keep_h=False, add_h=False) for smi in smis]

molecule_featurizer = featurizers.V1RDKit2DFeaturizer()
extra_mol_features = np.array([molecule_featurizer(mol) for mol in mols])

datapoints = [
    data.MoleculeDatapoint(mol, y, x_d=X_d)
    for mol, y, X_d in zip(
        mols,
        ys,
        extra_mol_features
    )
]

featurizer = featurizers.SimpleMoleculeMolGraphFeaturizer()

train_indices, val_indices, test_indices = data.make_split_indices(mols, "random", (0.8, 0.1, 0.1)) 
train_data, val_data, test_data = data.split_data_by_indices(
    datapoints, train_indices, val_indices, test_indices
)

train_dset = data.MoleculeDataset(train_data[0], featurizer)
val_dset = data.MoleculeDataset(val_data[0], featurizer)
test_dset = data.MoleculeDataset(test_data[0], featurizer)

scaler = train_dset.normalize_targets()
extra_datapoint_descriptors_scaler = train_dset.normalize_inputs("X_d")

val_dset.normalize_targets(scaler)
val_dset.normalize_inputs("X_d", extra_datapoint_descriptors_scaler)

num_workers = 5

train_loader = data.build_dataloader(train_dset, num_workers=num_workers)
val_loader = data.build_dataloader(val_dset, num_workers=num_workers, shuffle=False)
test_loader = data.build_dataloader(test_dset, num_workers=num_workers, shuffle=False)

batch_norm = True

mp = nn.BondMessagePassing()
agg = nn.MeanAggregation()

output_transform = nn.UnscaleTransform.from_standard_scaler(scaler)

ffn_input_dim = mp.output_dim + extra_mol_features.shape[1]

ffn = nn.RegressionFFN(input_dim=ffn_input_dim)
ffn = nn.RegressionFFN(input_dim=ffn_input_dim, output_transform=output_transform)

X_d_transform = nn.ScaleTransform.from_standard_scaler(extra_datapoint_descriptors_scaler)

metric_list = [nn.metrics.RMSE(), nn.metrics.MAE()]
mpnn = models.MPNN(mp, agg, ffn, metric_list, X_d_transform=X_d_transform)

trainer = pl.Trainer(
    logger=False,
    enable_checkpointing=True,
    enable_progress_bar=True,
    accelerator="auto",
    devices=1,
    max_epochs=2,
)

trainer.fit(mpnn, train_loader, val_loader)
results = trainer.test(dataloaders=test_loader)
```
Do I need to adapt my code for using the RDKit featurizers?

**Comment by KnathanM** [comment]
As a first check, what does `np.any(np.isnan(extra_mol_features ))` give you?

**Comment by tamarahalter** [comment]
There were indeed some NaNs in the extra_mol_features array. I removed them now but the error remains - I still get NaNs during the training. I also checked for inf values but there are none. 

```
extra_mol_features_cleaned = extra_mol_features[:, ~np.isnan(extra_mol_features).any(axis=0)]

datapoints = [
    data.MoleculeDatapoint(mol, y, x_d=X_d)
    for mol, y, X_d in zip(
        mols,
        ys,
        extra_mol_features_cleaned
    )
]
```

**Comment by KnathanM** [comment]
Based on what you describe, it still sounds like the problem is in the `extra_mol_features_cleaned` even though it has no nans or infs. Can you also try removing it entirely and seeing if you code works:
```
datapoints = [
    data.MoleculeDatapoint(mol, y)
    for mol, y in zip(
        mols,
        ys,
    )
]
```
Can you also test with a random numpy array instead?
```
datapoints = [
    data.MoleculeDatapoint(mol, y, x_d=X_d)
    for mol, y, X_d in zip(
        mols,
        ys,
        np.random.randn(*extra_mol_features_cleaned.shape)
    )
]
```

**Comment by tamarahalter** [comment]
In both cases the training works fine - no NaNs during the training

**Comment by KnathanM** [comment]
Okay good, I think that narrows down the issue to values inside `extra_mol_features`. Unfortunately, my only idea is to manually check all the values in that array, which may not be feasible if it is too large. I would check that the dtype is float and check again that there are no infs or nans. You could also check that it doesn't have None values. You could also try passing it through a single `torch.nn.Linear` layer to see if you get nans out. If you do, you could do a binary search for the bad value, but passing half the array through the linear layer, if it gives a nan, then pass half of that half array through the linear layer again until you can narrow down where in the original array is giving you trouble.

**Comment by JacksonBurns** [comment]
You may also want to consider normalizing the RDKit descriptors - if any of them are especially large, you could overflow the `float32` output.

Going to close this for the time being since it doesn't seem like a chemprop problem. Feel free to re-open if you find something amiss with chemprop that we can help with.
